ext/native/thin3d/VulkanRenderManager.cpp

```diff
@@ -368,6 +408,20 @@ void VulkanRenderManager::BeginFrame() {
 	vulkan_->BeginFrame();
 
 	insideFrame_ = true;
+
+	if (gpuProfilingEnabled_) {
+		// For various reasons, we need to always use an init cmd buffer in this case to perform the vkCmdResetQueryPool,
+		// unless we want to limit ourselves to only measure the main cmd buffer.
+		// Reserve the first two queries for initCmd.
+		frameData.numQueries = 2;
+		frameData.timestampDescriptions.push_back("initCmd Begin");
+		frameData.timestampDescriptions.push_back("initCmd");
+		VkCommandBuffer initCmd = GetInitCmd();
+		vkCmdResetQueryPool(initCmd, frameData.timestampQueryPool_, 0, MAX_TIMESTAMP_QUERIES);
+		vkCmdWriteTimestamp(initCmd, VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT, frameData.timestampQueryPool_, 0);
+	} else {
+		frameData.numQueries = 0;
```

Comment:
Hm, should we `frameData.timestampDescriptions.clear();` here?  And really before the other path too?

Actually, is there are scenario where we want `frameData.timestampDescriptions.size() != frameData.numQueries`?  It seems like we could just use the vector only.

-[Unknown]

Reply:
Yeah definitely, embarrassing mistake.

As for the separate variable, I had originally intended a description-less mode for a smaller performance hit but I don't think it's worth the trouble any more.